src/Migration/Reader/Map.php

```diff
@@ -141,12 +141,11 @@ public function isFieldDataTypeIgnored($document, $field, $type)
      */
     public function isDocumentIgnored($document, $type)
     {
-        $this->validateType($type);
         $key = $document . '-' . $type;
         if (isset($this->ignoredDocuments[$key])) {
             return $this->ignoredDocuments[$key];
         }
-
+        $this->validateType($type);
```

Comment:
If the result is already present in the `$this->ignoredDocuments`, then validation can be omitted